src/vscode-dts/vscode.proposed.chatSessionsProvider.d.ts

```diff
@@ -166,6 +171,32 @@ declare module 'vscode' {
 		 * @param token A cancellation token that can be used to cancel the operation.
 		 */
 		provideChatSessionContent(sessionId: string, token: CancellationToken): Thenable<ChatSession> | ChatSession;
+
+		/**
+		 *
+		 * @param sessionId Identifier of the chat session being updated.
+		 * @param updates Collection of option identifiers and their new values. Only the options that changed are included.
+		 * @param token A cancellation token that can be used to cancel the notification if the session is disposed.
+		 */
+		provideHandleOptionsChange?(sessionId: string, updates: ReadonlyArray<ChatSessionOptionUpdate>, token: CancellationToken): void;
```

Comment:
Can this live on the `ChatSession` as an optional `acceptOptionsChange` method?

Also I think we should type out what the options are instead of having a generic `ChatSessionOptionUpdate`

Reply:
I like this. It wasn't straightforward to me how to initially do this so lets talk about it more perhaps?  I left out the typing and curious what you think that should look like?